apps/src/blockly/addons/cdoFieldParameter.ts

```diff
@@ -0,0 +1,262 @@
+    if (variable) {
+      workspace.renameVariableById(variable.getId(), newName);
+
+      const procedureModel = (
+        definitionBlock as ProcedureBlock
+      ).getProcedureModel();
+      const parameters = procedureModel.getParameters();
+
+      // Find the parameter to rename
+      const paramToRename = parameters.find(
+        parameter => parameter.getName() === variableName
+      );
+
+      // Rename the parameter in the procedure model
+      if (paramToRename) {
+        paramToRename.setName(newName);
+      }
+    }
+  }
+  /**
+   * For a given parameter block, find the definition block and its workspace.
+   * If the block is part of the function definition, we want the root (top-most) block.
+   * If this block is in a mini-toolbox, we want the flyout's parent block.
+   * If neither of these are true (e.g. the block is disconnected on the workspace,
+   * we will attempt to find the first definition block on the same workspace that
+   * includes the parameter.
+   * @returns An object containing:
+   *   - `definitionBlock`: The definition block associated with the parameter
+   *   - `workspace`: The workspace that the parameter block belongs to.
+   */
+  protected findDefinitionBlockAndWorkspace(): {
+    definitionBlock: Block | null;
+    workspace: WorkspaceSvg;
+  } {
+    const parameterBlock = this.getSourceBlock() as Block;
+    let definitionBlock: Block | null;
+    let workspace = parameterBlock.workspace as WorkspaceSvg;
+
+    if (parameterBlock.isInFlyout) {
+      definitionBlock = (workspace as ExtendedWorkspaceSvg).flyoutParentBlock;
+      workspace = (workspace as WorkspaceSvg).targetWorkspace as WorkspaceSvg;
+    } else {
+      definitionBlock = parameterBlock.getRootBlock();
+    }
+
+    // Check if the root block is a definition block
+    if (
+      definitionBlock &&
+      definitionBlock.type !== BLOCK_TYPES.procedureDefinition
+    ) {
+      definitionBlock = null;
+    }
+
+    if (!definitionBlock) {
+      const variableName = this.getText();
+      const variable = workspace.getVariable(variableName);
+
+      if (variable) {
+        // Find the first block in the workspace that matches the expected type
+        definitionBlock =
+          workspace.getAllBlocks().find(block => {
+            if (block.type === BLOCK_TYPES.procedureDefinition) {
+              const procedureModel = (
+                block as ProcedureBlock
+              ).getProcedureModel();
+              return procedureModel
+                .getParameters()
+                .some(parameter => parameter.getName() === variableName);
+            }
+            return false;
+          }) || null;
+      }
+    }
+    return {definitionBlock, workspace};
+  }
+  /**
+   * Override of createTextArrow_ to fix the arrow position on Safari.
+   * We need to add dominant-baseline="central" to the arrow element in order to
+   * center it on Safari.
+   *  @override */
+  createTextArrow_() {
```

Comment:
This is copied directly from `CdoFieldVariable` and is here so that we don't have issue with dropdown arrows on any browsers. This repeated code is a good argument for "rebasing" this class as a follow-up.